### PR TITLE
feat: new dropdown

### PR DESCRIPTION
**Pull request by scmmishra**
This PR adds dropdown primitives to help compose custom dropdowns across the app. The following the sample usage

```vue
<DropdownContainer>
  <template #trigger="{ toggle }">
    <Button label="Open Menu" size="sm" @click="toggle" />
  </template>
  <DropdownBody class="w-80">
    <DropdownSection title="Profile Options">
      <DropdownItem label="Contact Support" class="justify-between">
        <span>{{ $t('SIDEBAR.SET_AUTO_OFFLINE.TEXT') }}</span>
        <div class="flex-shrink-0">
          <WootSwitch v-model="currentUserAutoOffline" />
        </div>
      </DropdownItem>
    </DropdownSection>
    <DropdownSeparator />
    <DropdownItem
      v-for="item in menuItems"
      :key="item.label"
      v-bind="item"
    />
  </DropdownBody>
</DropdownContainer>
```

### Preview

CleanShot 2024-11-18 at 13 32 17@2x

CleanShot 2024-11-18 at 13 32 24@2x

### CODE REVIEW

**Comment by scmmishra** [comment]
@iamsivin @pranavrajs need review on this

**Comment by pranavrajs** [comment]
@scmmishra Please take a look at this feedback. https://www.loom.com/share/2365090c4bcf4156beb79036ba01edf2. 

	1.	The feedback for the click action is somewhat slow.
	2.	The spacing between items doesn’t align with the gap between the actual sidebar items.

File: app/javascript/dashboard/components-next/sidebar/SidebarProfileMenuStatus.vue [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 import { useMapGetter, useStore } from 'dashboard/composables/store';
 import wootConstants from 'dashboard/constants/globals';
 import { useAlert } from 'dashboard/composables';
 import { useI18n } from 'vue-i18n';
 
+import DropdownSection from 'next/dropdown-menu/base/DropdownSection.vue';
+import DropdownItem from 'next/dropdown-menu/base/DropdownItem.vue';
+import DropdownContainer from 'next/dropdown-menu/base/DropdownContainer.vue';
+import DropdownBody from 'next/dropdown-menu/base/DropdownBody.vue';
+// import DropdownSeparator from 'next/dropdown-menu/base/DropdownSeparator.vue';
```
Comment (pranavrajs) [inline app/javascript/dashboard/components-next/sidebar/SidebarProfileMenuStatus.vue:12]: 👃

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Some minor feedback. Otherwise looks fine.

**Comment by scmmishra** [comment]
Hey @pranavrajs made the following changes in line with the feedback

1. Reduced the spacing around the dropdown items, made some layout tweaks with @absurdiya 
2. Updated the import across the dropdown component
3. We didn't have optimistic updates for the status changes previously, but it was straightforward to add, so made the changes for it. Let me know if you have any feedback about it :)

PTAL and let me know if this needs any changes. Thx.

<details><summary>P.S. Previous case (without optimistic updates)</summary>
<p>

Video from app.chatwoot.com with network throttling 

https://github.com/user-attachments/assets/b4f61872-3e2c-4c1b-88f0-8fe84e24773a

</p>
</details>

**Pull request merged**